scripts/src/java/org/oppia/android/scripts/maven/RetrieveLicenseTexts.kt

```diff
@@ -0,0 +1,413 @@
+  fun main(args: Array<String>) {
+    if (args.size < 2) {
+      println(
+        """
+        Usage: bazel run //scripts:generate_license_texts -- <path_to_directory_values>
+        <path_to_pb_file>  
+        """.trimIndent()
+      )
+      throw Exception("Too few arguments passed.")
+    }
+
+    val pathToValuesDirectory = args[0]
+    val pathToMavenDependenciesPb = args[1]
+    val valuesDirectory = File(pathToValuesDirectory)
+    check(valuesDirectory.isDirectory) { "Expected '$pathToValuesDirectory' to be a directory" }
+    val thirdPartyDependenciesXml = File(valuesDirectory, "third_party_dependencies.xml")
+
+    val mavenDependencyList = retrieveMavenDependencyList(pathToMavenDependenciesPb)
+    if (mavenDependencyList.isEmpty()) {
+      throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+    }
+    val copyrightLicenseSet = retrieveAllLicensesSet(mavenDependencyList)
+    val dependencyList = retrieveDependencyList(mavenDependencyList)
+
+    val dependencyNamesList = retrieveArtifactsNamesList(dependencyList)
+    val dependencyVersionsList = retrieveArtifactsVersionsList(dependencyList)
+
+    writeThirdPartyDependneciesXml(
+      thirdPartyDependenciesXml,
+      dependencyNamesList,
+      dependencyVersionsList,
+      copyrightLicenseSet,
+      dependencyList
+    )
+
+    println("\nScript execution completed successfully.")
+  }
+
+  /**
+   * Retrieve the list of maven dependencies from maven_dependencies.textproto.
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   * @return list of [MavenDependency]s
+   */
+  private fun retrieveMavenDependencyList(pathToPbFile: String): List<MavenDependency> {
+    return parseTextProto(
+      pathToPbFile,
+      MavenDependencyList.getDefaultInstance()
+    ).mavenDependencyList
+  }
+
+  /**
+   * Helper function to parse a binary proto file to a proto class.
+   *
+   * @param pathToPbFile path to the pb file to be parsed
+   * @param proto instance of the proto class
+   * @return proto class from the parsed textproto file
+   */
+  private fun parseTextProto(
+    pathToPbFile: String,
+    proto: MavenDependencyList
+  ): MavenDependencyList {
+    val protoBinaryFile = File(pathToPbFile)
+    val builder = proto.newBuilderForType()
+    return FileInputStream(protoBinaryFile).use {
+      builder.mergeFrom(it)
+    }.build() as MavenDependencyList
+  }
+
+  private fun retrieveAllLicensesSet(
+    mavenDependencyList: List<MavenDependency>
+  ): Set<CopyrightLicense> {
+    val copyrightLicensesSet = mutableSetOf<CopyrightLicense>()
+    mavenDependencyList.forEach { dependency ->
+      val licenseList = dependency.licenseList
+      if (licenseList.isEmpty()) {
+        throw Exception(MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE)
+      }
+      licenseList.forEach { license ->
+        copyrightLicensesSet.add(
+          retrieveCopyrightLicense(license)
+        )
+      }
+    }
+    return copyrightLicensesSet
```

Comment:
```suggestion
    return mavenDependencyList.flatMap { dependency ->
      val licenseList = dependency.licenseList
      check(licenseList.isNotEmpty()) { MAVEN_DEPENDENCY_LIST_NOT_UP_TO_DATE }
      return@flatMap licenseList.map { license ->
        retrieveCopyrightLicense(license)
      }
    }
```

This should be (effectively) equivalent, and avoids the mutability. Ditto elsewhere in this file that uses mutable lists (prefer to avoid them everywhere if we can).

Reply:
Done.